tutorials/math/tStudent.py

```diff
@@ -38,7 +38,12 @@
                                                               3.0))
 
 # For each quantile fill with the pdf
-xx = [-1.5] + [quant.GetBinContent(i)for i in range(1, 9)] + [1.5]
```

Comment:
Why eliminating the (nicer) list comprehension syntax?

Reply:
This is the same, but python 3 specific. The list comprehension in Python 3 defines the code that is run for each element as a function call, and hence in a new local scope. So it can not see the quant variable when run inside the makeimage wrapper.

Reply:
With the other proposed changes in this PR applied (so that python 3 is supported in the doxygen filter) except this rewrite of the list comprehension in the tStudent.py tutorial you get the following error when running the doxygen filter on the file:
```
$ DOXYGEN_SOURCE_DIRECTORY=../.. DOXYGEN_OUTPUT_DIRECTORY=/tmp/x PYTHON_EXECUTABLE=/usr/bin/python3 ./filter ../../tutorials/math/tStudent.py >/dev/null
[NbConvertApp] Converting notebook /tmp/x/notebooks/tStudent.py.ipynb to notebook
[NbConvertApp] Executing notebook with kernel: python3
[NbConvertApp] Writing 26589 bytes to /tmp/x/notebooks/tStudent.py.nbconvert.ipynb
Info in <ROOT::Math::MathMoreLibrary>: libMathMore has been loaded.
Traceback (most recent call last):
  File "makeimage.py", line 40, in <module>
    makeimage(argv[1], argv[2], argv[3], bool(argv[4]), bool(argv[5]), bool(argv[6]))
  File "makeimage.py", line 15, in makeimage
    if py: exec(compile(open(MacroName, "rb").read(), MacroName, 'exec'))
  File "../../tutorials/math/tStudent.py", line 41, in <module>
    xx = [-1.5] + [quant.GetBinContent(i)for i in range(1, 9)] + [1.5]
  File "../../tutorials/math/tStudent.py", line 41, in <listcomp>
    xx = [-1.5] + [quant.GetBinContent(i)for i in range(1, 9)] + [1.5]
NameError: name 'quant' is not defined
Segmentation fault (minnesutskrift skapad)
```
With the change it works as expected:
```
$ DOXYGEN_SOURCE_DIRECTORY=../.. DOXYGEN_OUTPUT_DIRECTORY=/tmp/x PYTHON_EXECUTABLE=/usr/bin/python3 ROOTSYS= ./filter ../../tutorials/math/tStudent.py >/dev/null
[NbConvertApp] Converting notebook /tmp/x/notebooks/tStudent.py.ipynb to notebook
[NbConvertApp] Executing notebook with kernel: python3
[NbConvertApp] Writing 26667 bytes to /tmp/x/notebooks/tStudent.py.nbconvert.ipynb
Info in <ROOT::Math::MathMoreLibrary>: libMathMore has been loaded.
TH1::TH1:0: RuntimeWarning: nbins is <=0 - set to nbins = 1
Info in <TCanvas::Print>: png file /tmp/x/html/pict1_tStudent.py.png has been created
```